### PR TITLE
Document 2.17 CA handling in KKP

### PR DESCRIPTION
**Pull request by xrstf**
This describes how https://github.com/kubermatic/kubermatic/pull/6538 would work out from an admin perspective.

### CODE REVIEW

File: content/kubermatic/master/guides/custom_ca/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+This document describes how to use a custom set of CA certificates in Kubermatic Kubernetes
+Platform (KKP). A general understanding of TLS certificates is assumed.
+
+KKP 2.17+ allows to configure a "CA bundle" (a set of CA certificates) on the master cluster.
+This CA bundle is then automatically
+
+* copied into each seed cluster
+* copied into each usercluster namespace
+* copied into each usercluster (into the `kube-system` namespace)
+* used for various components, like the KKP API, machine-controller, kube-apiserver etc.
```
Comment (xmudrii) [inline content/kubermatic/master/guides/custom_ca/_index.en.md:16]: Should we clarify that it's about the usercluster API server? For example:

```suggestion
* used for various components, like the KKP API, machine-controller, userclusters kube-apiserver, etc.
```

File: content/kubermatic/master/guides/custom_ca/_index.en.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+a starting point and then extended with your own CAs.
+
+KKP ships the CA bundle mentioned above by default in the `kubermatic-operator` Helm chart.
+This means that after installing KKP, a usable default CA bundle is available right from the
+start.
+
+To override the CA bundle, either overwrite the `static/ca-bundle.pem` file in the
+`kubermatic-operator` Helm chart with your own and then `helm upgrade` the operator, or manually
+change the ConfigMap later using `kubectl edit`. Changes to the ConfigMap are picked up
+automatically and are then reconciled.
```
Comment (xmudrii) [inline content/kubermatic/master/guides/custom_ca/_index.en.md:42]: If someone changes the ConfigMap and then runs `helm upgrade`, is Helm going to override the CA bundle?

File: content/kubermatic/master/guides/custom_ca/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    01yiPqFbQfXf5WRDLenVOavSot+3i9DAgBkcRcAtjOj4LaR0VknFBbVPFd5uRHg5h6h+u/N5GJG7
+    9G+dwfCMNYxdAfvDbbnvRG15RjF+Cv6pgsH/76tuIMRQyV+dTZsXjAzlAcmgQWpzU/qlULRuJQ/7
+    TBj0/VLZjmmx6BEP3ojY+x1J96relc8geMJgEtslQIxq/H5COEBkEveegeGTLg==
+    -----END CERTIFICATE-----
+
+    .... more certificates ....
+```
+
+The CA bundle needs to be configured in the `KubermaticConfiguration`; you need to configure
+the name of the ConfigMap in it:
```
Comment (xmudrii) [inline content/kubermatic/master/guides/custom_ca/_index.en.md:102]: As far as I got it, the ConfigMap is created by default, but what happens with the KubermaticConfiguration CR? Does it need to be updated? Will it use the ConfigMap by default without any update?

File: content/kubermatic/master/guides/custom_ca/_index.en.md (hunk 2 above)
Comment (xrstf) [reply content/kubermatic/master/guides/custom_ca/_index.en.md:42]: I am not sure, but I hope nobody will just flipflop between different ways of managing their CA bundle. :-D

File: content/kubermatic/master/guides/custom_ca/_index.en.md (hunk 3 above)
Comment (xrstf) [reply content/kubermatic/master/guides/custom_ca/_index.en.md]: Rephrased it a bit.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**